cockatrice/src/gamesmodel.cpp

```diff
@@ -91,32 +96,98 @@ QVariant GamesModel::data(const QModelIndex &index, int role) const
             switch (role) {
                 case Qt::DisplayRole: return prettyPrintSecsAgo(secs);
                 case SORT_ROLE: return QVariant(secs);
-                default: {
-                    qDebug() << "Returning data for col 1 of games model when role != display, role != sort";
-                    return QVariant(); // Shouldn't ever be reached.
-                }
+                default: return QVariant();
             }
         }
-        case 2: return QString::fromStdString(g.description());
-        case 3: return QString::fromStdString(g.creator_info().name());
-        case 4: {
-            QStringList result;
-            GameTypeMap gameTypeMap = gameTypes.value(g.room_id());
-            for (int i = g.game_types_size() - 1; i >= 0; --i)
-                result.append(gameTypeMap.value(g.game_types(i)));
-            return result.join(", ");
+        case DESCRIPTION: 
+            switch(role) {
+            case SORT_ROLE:
+            case Qt::DisplayRole:
+                return QString::fromStdString(g.description());
+            case Qt::TextAlignmentRole:
+                Qt::AlignLeft;
+            default:
+                return QVariant();
+        }
+        case CREATOR: {
+            switch(role) {
+            case SORT_ROLE:
+            case Qt::DisplayRole:
+                return QString::fromStdString(g.creator_info().name());
+            case Qt::DecorationRole:
+                switch(g.creator_info().user_level()) {
+                case 1: {
+                    QPixmap avatarPixmap = UserLevelPixmapGenerator::generatePixmap(13, (UserLevelFlags)g.creator_info().user_level());
+                    return QIcon(avatarPixmap);
+                        }
+                case 3:{
+                    QPixmap avatarPixmap = UserLevelPixmapGenerator::generatePixmap(13, (UserLevelFlags)g.creator_info().user_level());
+                    return QIcon(avatarPixmap);
```

Comment:
Is this the same block of code as case 1? You could fall through. Is there also no enum you can use? If not I'd comment what the cases are